app/tests/conftest.py

```diff
@@ -887,3 +890,145 @@ def submission_with_model_for_optional_inputs(jobs_for_optional_inputs):
         civs_for_interface2=jobs_for_optional_inputs.civs_for_interface2,
         output_civs=jobs_for_optional_inputs.output_civs,
     )
+
+
+class PhaseWithInputs(NamedTuple):
```

Comment:
Large fixtures like this should be avoided if possible as you can create weird dependencies between tests. The fixture anyway does not need to live in conftest where it is available to everything and can instead be confined to the module where it is used.

Reply:
I will move it to the evaluation tests. Repeating all the set-up for every test that uses this also seems suboptimal. What would you suggest doing instead?

Reply:
Yes sometimes it's necessary, but usually it is better to limit the number of things that are created for a test. Otherwise things get slow as you're creating a lot of (potentially unnecessary) objects, and if you need to change the fixture you do not know if you are breaking an implicit assumption of a test that is using it.

For this reason we moved away from things like the two challenge sets, and wrote tests that were more isolated instead, or just duplicated a lot of factories.